src/test/java/tahafurkan/sandbox/usermanagement/service/UserServiceTest.java

```diff
@@ -92,4 +101,53 @@ void create_GivenAllowedUsername_ReturnsSavedUser() {
         Mockito.verify(userRepository, times(1)).save(user);
     }
 
+    @Test
+    void get_GivenIdFind_ThrowsException() {
+        assertThrows(NoSuchUserExistsException.class, () -> {
+            int id = 1;
+            User user = new User();
+            user.setId(id);
+            userService.get(id);
+        });
+    }
+
+    @Test
+    void getAll_ReturnsUser() {
+        userService.getAll();
+        Mockito.verify(userRepository).findAll();
+    }
+
+    @Test
+    void delete_GivenIdDelete_ThrowsException() {
+        assertThrows(NoSuchUserExistsException.class, () -> {
+            int id = 1;
+            userService.delete(id);
+        });
+    }
+
+    @Test
+    void getByUsername_GivenFindUsername_ReturnUser() {
+        String username = "taha.furkan";
+        User user = new User();
+        user.setUsername(username);
+        given(userRepository.findByUsername(username)).willReturn(user);
+        userService.getByUsername(username);
+    }
+    @Test
+    void update_GivenIdUpdate_ReturnUser(){
```

Comment:
State under test olarak verdigin `_GivenIdUpdate_` bir sey ifade etmiyor. Test metodunda kontrollu olarak varolan `User`'in bir alanini guncelle, mesela `last name`'i guncelleyen bir test olsun bu.

`update_UpdatesLastName_ReturnsUpdatedUser` seklinde bir isimlendirme kullanabilirsin o durumda.